pkg/storage/allocator_scorer.go

```diff
@@ -541,19 +544,20 @@ func shouldRebalance(
 	store roachpb.StoreDescriptor,
 	sl StoreList,
 	rangeInfo RangeInfo,
+	disableStatsBasedRebalance bool,
 ) bool {
 	if store.Capacity.FractionUsed() >= maxFractionUsedThreshold {
 		log.VEventf(ctx, 2, "s%d: should-rebalance(disk-full): fraction-used=%.2f, capacity=(%v)",
 			store.StoreID, store.Capacity.FractionUsed(), store.Capacity)
 		return true
 	}
 
-	if !statsBasedRebalancingEnabled(st) {
+	if !statsBasedRebalancingEnabled(st) || disableStatsBasedRebalance {
```

Comment:
All these different conditionals would be less error prone if we just passed `disableStatsBasedRebalancing` into `statsBasedRebalancingEnabled`.

Reply:
Done.